src/etc/inc/services.inc

```diff
@@ -1190,6 +1193,18 @@ EOD;
 				}
 
 				$dhcpdconf .= "}\n";
+				// add zone DDNS key/server to $ddns_zone[] if required
+				if ($need_sm_ddns_updates) {
+					$sm_newzone['dns-servers'] = array($sm['ddnsdomainprimary']);
+					$sm_newzone['domain-name'] = $sm['domain'];
```

Comment:
If multiple static maps use the same domain, won't this make multiple entries for the same domain? Is that allowed and functional in ISC DHCPD?

Reply:
I was thinking about this as well, but even though a new zone-entry for `$ddns_zones[]` is added multiple times, it will get filtered by `function dhcpdzones($ddns_zones)`:
```
function dhcpdzones($ddns_zones) {
...
                        // We don't need to add zones multiple times.
                        if ($zone['domain-name'] && !in_array($zone['domain-name'], $added_zones)) {
...
```

Reply:
If it will be skipped, then we should not bother to add it in the first place, rather than stuffing the array full of potentially redundant information. Also if it's duplicated across multiple interfaces/entries then it probably needs to be made clear to the user that only the first defined set of options per domain will be honored.

Reply:
In case the information entered on a per-node basis is not added to that array, it will be skipped entirely. We could add a check if an entry is already pushed to that array, but it would be duplicated code. If you come up with a better solution, I'm fine with it. The current implementation offers options, which are simply ignored as of now.